packages/@aws-cdk/aws-servicecatalog/lib/template.ts

```diff
@@ -0,0 +1,93 @@
+import * as s3_assets from '@aws-cdk/aws-s3-assets';
+import * as cdk from '@aws-cdk/core';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * Represents the Product Provisioning Artifact Template.
+ */
+export abstract class Template {
+  /**
+   * Template from URL
+   * @param url The url that points to the provisioning artifacts template
+   */
+  public static fromUrl(url: string): UrlTemplate {
```

Comment:
Let's actually make this method return `Template` instead, and let's keep `UrlTemplate` module-private (not exported).